app/lib/work_history_duration.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class WorkHistoryDuration
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def count_months
+    work_histories
+      .map { |work_history| work_history_full_time_months(work_history) }
+      .sum
+      .round
+  end
+
+  private
+
+  AVERAGE_WEEKS_PER_MONTH = 4.34
+  HOURS_PER_FULL_TIME_MONTH = 130.0
+
+  attr_reader :application_form
+
+  def work_histories
+    @work_histories ||=
+      application_form
+        .work_histories
+        .where.not(start_date: nil)
```

Comment:
Could these be a scope on WorkHistory?

Reply:
I'm not sure, it feels quite specific to this class to me, in the sense that the only values we _need_ to calculate the duration are the `start_date` and the `hours_per_week`. I could add a `has_start_date` and `has_hours_per_week` scope, but that doesn't feel worth it to me, and having a `can_calculate_duration` scope feels like leaking responsibilities of the duration class to the model. What do you think?